File: app/src/main/java/com/kode/utair/data/network/WeatherApiService.java
package com.kode.utair.data.network;

import com.kode.utair.data.network.responses.WeatherForecastResponse;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface WeatherApiService {

    String BASE_URL = "http://api.openweathermap.org/data/2.5/";
    String API_KEY = "eb83085d6324416a2af66f8aa214eff8";
    String CELSIUS_METRICS_KEY = "metric";

    @GET("forecast?units=" + CELSIUS_METRICS_KEY)
    Single<WeatherForecastResponse> getWeather(@Query("q") String cityName);


}

File: app/src/main/java/com/kode/utair/data/network/WeatherApiService.kt
package com.kode.utair.data.network

import com.kode.utair.BuildConfig
import com.kode.utair.data.network.responses.WeatherForecastResponse
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface WeatherApiService {

    companion object {
        const val BASE_URL = "http://api.openweathermap.org/data/2.5/"
        const val API_KEY = BuildConfig.OPENWEATHER_API_KEY
        const val CELSIUS_METRICS_KEY = "metric"
    }

    @GET("forecast?units=$CELSIUS_METRICS_KEY")
    fun getWeather(@Query("q") cityName: String?): Single<WeatherForecastResponse?>?

}